proposals/299-ip-failure-count.txt

```diff
@@ -191,14 +195,13 @@ Ticket: https://trac.torproject.org/projects/tor/ticket/27491
    packets for a particular IP family while still preventing potential denial
    of service attacks.
 
-8. Pathbias and Failure Probability
+8. Marking Guards As Failed with ClientAutoIPv6ORPort
+
+   If ClientAutoIPv6ORPort is in use, we should mark a guard as failed if we
+   fail both IPv4 and IPv6 connections to the guard.
```

Comment:
Here are my previous comments:

> *When Tor is using both IPv4 and IPv6*, we should only mark as guard as failed if both IPv4 and IPv6 fail to that guard.
>
> We could try to work out which options make Tor use IPv4 and IPv6, I think they are ClientAutoIPv6ORPort, ClientPreferIPv6ORPort, and configured IPv4 and IPv6 bridges. Or we could just count the total number of attempts on IPv4 and IPv6 (and divide by 2 to forget history like we do with failures).

I don't think that checking ClientAutoIPv6ORPort is the right way to check if tor is using IPv4 and IPv6. What about ClientPreferIPv6ORPort, ClientUseIPv6ORPort, and configured IPv4 and IPv6 bridges?

We should fix the guard failure bug for all of these cases, not just ClientAutoIPv6ORPort.

Reply:
>I don't think that checking ClientAutoIPv6ORPort is the right way to check if tor is using IPv4 and IPv6. What about ClientPreferIPv6ORPort, ClientUseIPv6ORPort, and configured IPv4 and IPv6 bridges?

Makes sense.

>We should fix the guard failure bug for all of these cases, not just ClientAutoIPv6ORPort.

If you are okay with this change, I will change the proposal to say that if **any** option like ClientAutoIPv6ORPort, ClientPreferIPv6ORPort, or ClientUseIPv6ORPort is set, or if we have both IPv4 and IPv6 bridges set, we do this.

Reply:
I don't think we should check options: if we do, we need to remember to update the code each time we change the options.

Instead, I think we should look at the connections that Tor is actually attempting. If Tor has recently succeeded on both IPv4 and IPv6, we should require failures on both protocols to fail a guard. If Tor has only recently succeeded on one protocol, we should only require that protocol.

Reply:
Makes perfect sense.